### PR TITLE
AccessApi for fileshare POST, GET , PUT, DELETE

### PR DESCRIPTION
**Pull request by PravinRanjan10**
**What this PR does / why we need it**:
It stores the metadata of access rule for fileshare based
**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
```

### CODE REVIEW

File: pkg/api/controllers/fileshare.go [hunk 1]
```diff
@@ -37,7 +37,59 @@ type FileSharePortal struct {
...
 	CtrClient client.Client
 }
+// Function to store Acl's related entry into databse
+func (f *FileSharePortal) CreateFileShareAcl() {
+	var fileshareacl = model.FileShareAclSpec{
+		BaseModel: &model.BaseModel{},
+	}
+	// Unmarshal the request body
+	if err := json.NewDecoder(f.Ctx.Request.Body).Decode(&fileshareacl); err != nil {
+		reason := fmt.Sprintf("Parse fileshare access rules request body failed: %s", err.Error())
```
Comment (wisererik) [inline pkg/api/controllers/fileshare.go]: lowercase

File: pkg/api/controllers/fileshare.go [hunk 2]
```diff
@@ -37,7 +37,59 @@ type FileSharePortal struct {
...
+		f.ErrorHandle(model.ErrorBadRequest, reason)
+		log.Error(reason)
+		return
+	}
+	result, err := util.CreateFileShareAclDBEntry(c.GetContext(f.Ctx), &fileshareacl)
+	if err != nil {
+		reason := fmt.Sprintf("Create access rules for fileshare failed: %s", err.Error())
+		f.ErrorHandle(model.ErrorBadRequest, reason)
```
Comment (wisererik) [inline pkg/api/controllers/fileshare.go]: lowercase for error log

File: testutils/collection/data.go [hunk 3]
```diff
@@ -165,6 +165,22 @@ var (
 			SnapshotId:  "a5965ebe-dg2c-434t-b28e-f373746a71ca",
 		},
 	}
+
+	SampleFileSharesAcl = []model.FileShareAclSpec{
+		{
+			BaseModel: &model.BaseModel{
+				Id: "d2975ebe-d82c-430f-b28e-f373746a71ca",
+			},
+			Description: "This is a sample fileshare for testing",
```
Comment (wisererik) [inline testutils/collection/data.go]: change description

**Comment by xing-yang** [comment]
@PravinRanjan10 It looks like you created a new repo under the opensds org by accident: https://github.com/opensds/AccessApi.  I'm going to delete that repo.  Please save your work.

**Comment by xing-yang** [comment]
Ok.  Will do.  Thanks.

File: pkg/api/controllers/fileshare.go (hunk 1 above)
Comment (PravinRanjan10) [reply pkg/api/controllers/fileshare.go]: done

File: pkg/db/drivers/etcd/etcd.go [hunk 4]
```diff
@@ -257,6 +257,27 @@ func (c *Client) FindFileShareValue(k string, p *model.FileShareSpec) string {
...
+		log.Error("when create fileshare access rules in db:", dbRes.Error)
+		return nil, errors.New(dbRes.Error)
+	}
+
+	return fshare, nil
+}
+
 func (c *Client) CreateFileShare(ctx *c.Context, fshare *model.FileShareSpec) (*model.FileShareSpec, error) {
 	//profiles, err := c.ListProfiles(ctx)
 	//if err != nil {
```
Comment (wisererik) [inline pkg/db/drivers/etcd/etcd.go]: remove unused codes

File: pkg/api/util/db.go [hunk 5]
```diff
@@ -33,6 +33,35 @@ import (
...
+
+	in.Description = in.Description
+
+	in.Type = in.Type
+	in.AccessTo = in.AccessTo
+	in.AccessCapability = in.AccessCapability
+	_, err := db.C.GetFileShare(ctx, in.FileShareId)
+	if err != nil {
+		log.Error("FileshareID is not valid: ", err)
+		return nil, err
```
Comment (wisererik) [inline pkg/api/util/db.go]: FilesareID -> file share id

File: pkg/api/util/db.go [hunk 6]
```diff
@@ -33,6 +33,35 @@ import (
 	uuid "github.com/satori/go.uuid"
 )
 
+//function to store filesahreAcl metadata into database
```
Comment (xing-yang) [inline pkg/api/util/db.go]: nit: add space after //

**Pull request merged**